op-node/rollup/derive/channel_out.go

```diff
@@ -68,7 +70,7 @@ func NewChannelOut(batchType uint, compress Compressor, spanBatch *SpanBatch) (C
 	case SingularBatchType:
 		return NewSingularChannelOut(compress)
 	case SpanBatchType:
-		return NewSpanChannelOut(compress, spanBatch)
+		return NewSpanChannelOut(spanBatch.GenesisTimestamp, spanBatch.ChainID, compress.TargetOutputSize())
 	default:
 		return nil, fmt.Errorf("unrecognized batch type: %d", batchType)
 	}
```

Comment:
I can refactor this, sure. It's not so straight forward though, for two reasons:
* `NewChannelOut` is used in tests to test Span and Singular Batches with the same test cases. That's okay because they can still wrap this logic up
* If there's not a compressor to consult, we still must determine a `TargetOutput`. (EDIT: actually that's not as bad as I first thought)